docsite/source/index.html.md

```diff
@@ -0,0 +1,97 @@
+---
+title: Introduction
+layout: gem-single
+type: gem
+name: dry-operation
+sections:
+  - error-handling
+  - configuration
+  - extensions
+  - design-pattern
+---
+
+dry-operation provides an expressive and flexible way for you to model your app's business operations. It provides a lightweight DSL around [dry-monads](/gems/dry-monads/), which allows you to chain together steps and operations with a focus on the happy path, while elegantly handling failures.
+
+### Introduction
+
+In complex business logic, it's common to have a series of operations that depend on each other. Traditionally, this leads to deeply nested conditional statements or a series of guard clauses. dry-operation provides a more elegant solution by allowing you to define a linear flow of operations, automatically short-circuiting on failure.
+
+### Basic Usage
+
+To use dry-operation, create a class that inherits from `Dry::Operation` and define your flow in the `#call` method:
+
+```ruby
+class CreateUser < Dry::Operation
+  def call(input)
+    attrs = step validate(input)
+    user = step persist(attrs)
+    step notify(user)
+    user
+  end
+
+  private
+
+  def validate(input)
+    # Return Success(attrs) or Failure(error)
+  end
+
+  def persist(attrs)
+    # Return Success(user) or Failure(error)
+  end
+
+  def notify(user)
+    # Return Success(true) or Failure(error)
+  end
+end
+```
+
+Each step (`validate`, `persist`, `notify`) is expected to return either a `Success` or `Failure` from [dry-monads](/gems/dry-monads/).
+
+### The step method
+
+The `#step` method is the core of `Dry::Operation`. It does two main things:
+
+- If the result is a `Success`, it unwraps the value and returns it.
+- If the result is a `Failure`, it halts the execution throwing the failure up the call stack.
```

Comment:
"halts the execution throwing the failure up the call stack" feels like too technical and hard to understand.

In the error-handling page, we have something that feels a lot better to me:

> When a step returns a `Failure`, the operation short-circuits, skipping the remaining steps and returning the failure immediately.

Do you reckon you might be able to adjust this to be a little easier to understand?

Reply:
```
- If the result is a `Failure`, it short-circuits the operation and returns the failure.
```

Let me know what you think 🙂

Reply:
Perfect!